flutter/lib/src/widgets_binding_observer.dart

```diff
@@ -20,11 +24,31 @@ class SentryWidgetsBindingObserver with WidgetsBindingObserver {
     Hub? hub,
     required SentryFlutterOptions options,
   })  : _hub = hub ?? HubAdapter(),
-        _options = options;
+        _options = options,
+        _screenSizeStreamController = StreamController(sync: true) {
+    if (_options.enableWindowMetricBreadcrumbs) {
+      _screenSizeStreamController.stream
+          .map(
+            (window) => {
+              'new_pixel_ratio': window?.devicePixelRatio,
+              'new_height': window?.physicalSize.height,
+              'new_width': window?.physicalSize.width,
+            },
+          )
+          .distinct(mapEquals)
+          .skip(1) // Skip initial event added below in constructor
+          .listen(_onScreenSizeChanged);
+
+      final window = BindingUtils.getWidgetsBindingInstance()?.window;
+      _screenSizeStreamController.add(window);
+    }
+  }
 
   final Hub _hub;
   final SentryFlutterOptions _options;
 
+  final StreamController<SingletonFlutterWindow?> _screenSizeStreamController;
```

Comment:
I don't use `deepEquals`, there is actually a `mapEquals` function used instead, which provides shallow check with top level keys comparison only, but i like the idea with ViewConfiguration, i'll try it 😄

Reply:
@ueman I've just checked, it's not worth to rely on `ViewConfiguration` and his equals method, since it also contains `viewInsets` & other fields besides screen size & pixel ratio